.github/workflows/actions/build-version/action.yml

```diff
@@ -0,0 +1,135 @@
+name: Build package
+inputs:
+  crate:
+    description: crates to build
+    required: true
+    type: string
+  features:
+    description: features to enable
+    required: true
+    type: string
+  no_default_features:
+    description: disable default features
+    required: true
+    type: string
+  target_arch:
+    description: Target to build against
+    required: true
+    type: string
+  version:
+    description: Version to release
+    required: true
+    type: string
+
+outputs:
+  artifact:
+    description: "artifact"
+    value: ${{ steps.output.outputs.artifact }}
+
+runs:
+  using: composite
+  steps:
+    - uses: taiki-e/install-action@v2
+      if: runner.os != 'Linux'
+      with:
+        tool: cargo-auditable
+
+    - name: Install LLVM
+      if: runner.os == 'Windows'
+      run: choco install llvm
+      shell: bash
+
+    - name: Install LLVM
+      if: runner.os == 'macOS'
+      run: |
+        set -euxo pipefail
+        brew install llvm
+        LLVM_PREFIX="$(brew --prefix llvm)"
+        echo "${LLVM_PREFIX}/bin" >> "$GITHUB_PATH"
+        echo "LLVM_CONFIG_PATH=${LLVM_PREFIX}/bin/llvm-config" >> "$GITHUB_ENV"
+        echo "DYLD_LIBRARY_PATH=${LLVM_PREFIX}/lib:${LD_LIBRARY_PATH:-}" >> "$GITHUB_ENV"
+        echo "LIBCLANG_PATH=$(llvm-config --libdir)" >> "$GITHUB_ENV"
+        echo "LLVM_DIR=$(llvm-config --cmakedir)" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Decide cargo args, tempdir and cargo root
+      env:
+        FEATURES: ${{ inputs.features }}
+        NO_DEFAULT_FEATURES: ${{ inputs.no_default_features }}
+      run: |
+        set -exuo pipefail
+        features="${FEATURES:-}"
+        if [ -z "$features" ]; then
+          feature_flag=""
+        else
+          feature_flag="--features"
+        fi
+        no_default_features=""
+        if [ "${NO_DEFAULT_FEATURES:-}" = "true" ]; then
+          no_default_features='--no-default-features'
+        fi
+        echo "CARGO_ARGS=$no_default_features $feature_flag $features" >> "$GITHUB_ENV"
+        echo "TEMPDIR=$(mktemp -d)" >> "$GITHUB_ENV"
+        echo CARGO_HOME="$(mktemp -d 2>/dev/null || mktemp -d -t 'cargo-root')" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Build packages
+      if: runner.os != 'Linux'
+      run: ./build-version.sh
```

Comment:
> > What is the threat model that this is mitigating against?
>
> I think each job in GitHub actions still have access to some environment variables (implicitly defined) and some github context

If getting access to these things is game-over then we should turn off all non-linux builders and cross-compile everything from linux or something. I really hope it's not though (read https://alsuren.github.io/2022/07/10/cargo-quickinstall.html#github-actions-gymnastics for some history about why I'm treating it as basically fine)

Reply:
@NobodyXu let's make sure we've addressed this thread too.

Reply:
I think this is related to what we discussed in https://github.com/cargo-bins/cargo-quickinstall/pull/252#discussion_r1645007351

Regarding cross compilation, I think zig-cc could help us achieve cross compiling to macos and mingw-gnu on Linux.